oak_functions_service/src/wasm/wasmtime.rs

```diff
@@ -0,0 +1,522 @@
+//
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+//! Wasm business logic provider based on [wasmtime](https://wasmtime.dev/).
+//!
+//! This file is mostly a copy of the file `mod.rs`, replacing `wasmi` with `wasmtime`, and fixing
+//! some minor issues. The API of `wasmi` and `wasmtime` are remarkably similar, so the changes are
```

Comment:
I was hoping it would just be a case of defining different type aliases based on a feature flag, but I guess the minor issues stopped that simple approach.